Index: custom_pagers.module
===================================================================
--- custom_pagers.module	(revision 114)
+++ custom_pagers.module	(working copy)
@@ -259,7 +259,9 @@
     if (!empty($pager->args)) {
       $args = array_map('trim', explode("\n", $pager->args));
       if (module_exists('token')) {
-        $args = token_replace($args, 'node', $node);
+        foreach($args as $key => $value) {
+          $args[$key] = token_replace($value, 'node', $node);
+        }
       }
     }
     else {
